File: backend/app/models/telegram.py
from sqlalchemy import Column, Integer, String, DateTime, Text, Boolean
from datetime import datetime
from app.core.database import Base


class TelegramMessage(Base):
    __tablename__ = "telegram_messages"

    id = Column(Integer, primary_key=True, index=True)
    chat_id = Column(String(64), nullable=False, index=True)
    username = Column(String(128), nullable=True)
    direction = Column(String(8), nullable=False)  # "in" | "out"
    text = Column(Text, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow)
    is_command = Column(Boolean, default=False)
    file_id = Column(String(256), nullable=True)
    file_type = Column(String(32), nullable=True)  # photo | document | video | sticker

File: backend-go/models/telegram.go
package models

import "time"

type TelegramMessage struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	ChatID    string    `gorm:"size:64;not null;index" json:"chat_id"`
	Username  *string   `gorm:"size:128" json:"username"`
	Direction string    `gorm:"size:8;not null" json:"direction"`
	Text      string    `gorm:"type:text;not null" json:"text"`
	CreatedAt time.Time `json:"created_at"`
	IsCommand bool      `gorm:"default:false" json:"is_command"`
	FileID    *string   `gorm:"size:256" json:"file_id"`
	FileType  *string   `gorm:"size:32" json:"file_type"`
}

func (TelegramMessage) TableName() string { return "telegram_messages" }
